service/src/main/java/uk/gov/hmcts/reform/fpl/service/PrepareDirectionsForUsersService.java

```diff
@@ -119,17 +120,26 @@ private boolean notCompliedWithByCourt(String onBehalfOf, Element<DirectionRespo
                                                   List<Element<Direction>> directions) {
         return directions.stream()
             .map(element -> element(element.getId(), element.getValue().toBuilder()
-                .response(element.getValue().getResponses().stream()
-                    .filter(response -> response.getValue().getDirectionId().equals(element.getId()))
-                    .filter(response -> response.getValue().getAssignee().equals(assignee))
-                    .map(Element::getValue)
-                    .findFirst()
-                    .orElse(null))
+                .response(getResponse(assignee, element))
                 .responses(emptyList())
                 .build()))
             .collect(toList());
     }
 
+    private DirectionResponse getResponse(DirectionAssignee assignee, Element<Direction> element) {
+        return element.getValue().getResponses().stream()
+            .filter(response -> response.getValue().getDirectionId().equals(element.getId()))
+            .filter(response -> {
+                if ("CAFCASS".equals(response.getValue().getRespondingOnBehalfOf())) {
+                    return response.getValue().getRespondingOnBehalfOf().equals(assignee.toString());
+                }
+                return response.getValue().getAssignee().equals(assignee);
```

Comment:
Looks like current unit tests doesn't cover this scenario, is there a benefit to add a test something like `shouldExtractPartyResponsesForGivenPartyWithCorrectAssignee` ? (if i understood this correctly)

Reply:
new test is there for ya!